spec/namespaces/geospatial.yaml

```diff
@@ -0,0 +1,246 @@
+      x-operation-group: geospatial.ip_to_geo_get_all_datasource
+      x-version-added: '2.11'
+      description: Get information about all IP2Geo data sources.
+      externalDocs:
+        url: 'https://docs.opensearch.org/docs/latest/ingest-pipelines/processors/ip2geo/'
+      responses:
+        '200':
+          $ref: '#/components/responses/geospatial.ip_to_geo_get_all_datasource@200'
+
+  /_plugins/geospatial/ip2geo/datasource/{datasource_name}:
+    get:
+      operationId: geospatial.ip_to_geo_get_datasource.0
+      x-operation-group: geospatial.ip_to_geo_get_datasource
+      x-version-added: '2.11'
+      description: Get information for a specific IP2Geo data source.
+      externalDocs:
+        url: 'https://docs.opensearch.org/docs/latest/ingest-pipelines/processors/ip2geo/#sending-a-get-request'
+      parameters:
+        - $ref: '#/components/parameters/geospatial.ip_to_geo_get_datasource::path.datasource_name'
+      responses:
+        '200':
+          $ref: '#/components/responses/geospatial.ip_to_geo_get_datasource@200'
+    put:
+      operationId: geospatial.ip_to_geo_put_datasource.0
+      x-operation-group: geospatial.ip_to_geo_put_datasource
+      x-version-added: '2.11'
+      description: |-
+        Create a specific IP2Geo data source.
+        Default values:
+          - `endpoint`: `"https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json"`
+          - `update_interval_in_days`: 3.
+      externalDocs:
+        url: 'https://docs.opensearch.org/docs/latest/ingest-pipelines/processors/ip2geo/#data-source-options'
+      parameters:
+        - $ref: '#/components/parameters/geospatial.ip_to_geo_put_datasource::path.datasource_name'
+      requestBody:
+        $ref: '#/components/requestBodies/geospatial.ip_to_geo_put_datasource'
+      responses:
+        '200':
+          $ref: '#/components/responses/geospatial.ip_to_geo_put_datasource@200'
+        '400':
+          $ref: '#/components/responses/geospatial.ip_to_geo_put_datasource@400'
+    delete:
+      operationId: geospatial.ip_to_geo_delete_datasource.0
+      x-operation-group: geospatial.ip_to_geo_delete_datasource
+      x-version-added: '2.11'
+      description: Delete a specific IP2Geo data source.
+      externalDocs:
+        url: 'https://docs.opensearch.org/docs/latest/ingest-pipelines/processors/ip2geo/#deleting-the-ip2geo-data-source'
+      parameters:
+        - $ref: '#/components/parameters/geospatial.ip_to_geo_delete_datasource::path.datasource_name'
+      responses:
+        '200':
+          $ref: '#/components/responses/geospatial.ip_to_geo_delete_datasource@200'
+        '404':
+          $ref: '#/components/responses/geospatial.ip_to_geo_delete_datasource@404'
+  /_plugins/geospatial/ip2geo/datasource/{datasource_name}/_settings:
+    put:
+      operationId: geospatial.ip_to_geo_datasource_settings.0
+      x-operation-group: geospatial.ip_to_geo_datasource_settings
```

Comment:
In case there's a chance of a get variant in future, this should ideally be `put_ip_to_geo_datasource_settings`/`put_ip2geo_datasource_settings`.